tests/functional/test_crt.py

```diff
@@ -0,0 +1,231 @@
+import mock
+import unittest
+import threading
+import re
+from concurrent.futures import Future
+
+from botocore.session import Session
+
+from s3transfer.subscribers import BaseSubscriber
+
+from tests import FileCreator
+from tests import requires_crt, HAS_CRT
+
+if HAS_CRT:
+    import s3transfer.crt
+    import awscrt
+
+
+class submitThread(threading.Thread):
+    def __init__(self, transfer_manager, futures, callargs):
+        threading.Thread.__init__(self)
+        self._transfer_manager = transfer_manager
+        self._futures = futures
+        self._callargs = callargs
+
+    def run(self):
+        self._futures.append(self._transfer_manager.download(*self._callargs))
+
+
+class SerializationException(Exception):
+    pass
+
+
+if HAS_CRT:
+    class ExceptionRaisingSerializer(s3transfer.crt.BaseCRTRequestSerializer):
+        def serialize_http_request(self, transfer_type, future):
+            raise SerializationException()
```

Comment:
Since we only use this class once I was thinking we just move it into the test case. So something like this to reduce the number of `if HAS_CRT` checks:
```python
    def test_serializer_error_handling(self):
        class SerializationException(Exception):
             pass

       class ExceptionRaisingSerializer(s3transfer.crt.BaseCRTRequestSerializer):
           def serialize_http_request(self, transfer_type, future):
               raise SerializationException()


        not_impl_serializer = ExceptionRaisingSerializer()
        transfer_manager = s3transfer.crt.CRTTransferManager(
            crt_s3_client=self.s3_crt_client,
            crt_request_serializer=not_impl_serializer)
        future = transfer_manager.upload(
            self.filename, self.bucket, self.key, {}, [])

        with self.assertRaises(SerializationException):
            future.result()
```